iradix.go

```diff
@@ -183,6 +183,32 @@ func (t *Txn) writeNode(n *Node, forLeafUpdate bool) *Node {
 	return nc
 }
 
+// Visit all the nodes in the tree under n, and add their mutateChannels to the transaction
+func (t *Txn) recursiveWatchNotify(n *Node, fn func(n *Node) bool) bool {
+
+	// Mark this node as being mutated.
+	if t.trackMutate {
+		t.trackChannel(n.mutateCh)
+	}
+
+	// Mark its leaf as being mutated, if appropriate.
+	if t.trackMutate && n.leaf != nil {
+		t.trackChannel(n.leaf.mutateCh)
+	}
+
+	if n.leaf != nil && fn(n) {
```

Comment:
I started this modelling some other walk functions I saw in the code base, agree now that this is too confusing and doesn't need to be implemented this way. I'll simplify it